controllers/controllers.go

```diff
@@ -49,19 +49,27 @@ func metadataMatches(currentMetadata metav1.ObjectMeta, desiredMetadata metav1.O
 	return containsAll(currentMetadata.Labels, desiredMetadata.Labels) && containsAll(currentMetadata.Annotations, desiredMetadata.Annotations)
 }
 
-// mergeLabels merges the the labels specified by the operator into
+// mergeLabels merges the labels specified by the operator into
 // on object's metadata.
 //
 // This will return whether the target's labels have changed.
 func mergeLabelsInMetadata(target *metav1.ObjectMeta, desired metav1.ObjectMeta) bool {
+	if target.Labels == nil && desired.Labels != nil && len(desired.Labels) > 0 {
```

Comment:
```suggestion
	if target.Labels == nil  {
```

We don't really care about the desired in this case, as we only set the `Labels` to an empty map in case that the `Labels` are nil. If `desired.Labels` is nil or empty, the for loop in `mergeMap` will just be a no-op (as there are no items to iterate over). Is the intention here to only initialize the map if the desired labels contain at least one element? If so, we could simplify it to `target.Labels == nil && len(desired.Labels) > 0` (the `len` method will accept a nil map).

Reply:
> Is the intention here to only initialize the map if the desired labels contain at least one element?

Yes; other tests check that the target map remains `nil` if it's empty. I'll simplify the check, though!